app/assets/javascripts/multi_image_upload.js.erb

```diff
@@ -1,779 +1,793 @@
-        });
-
-        // IMPORTANT:  This allows the user to update the thumbnail on the edit
-        // observation view.
-        jQuery('input[type="radio"][name="observation[thumb_image_id]"]').change(function() {
-            jQuery('#observation_thumb_image_id').val($(this).val());
-        });
+      event.preventDefault();
 
-        // Logic for setting the default thumbnail
-        jQuery('body').on('click', '[data-role="set_as_default_thumbnail"]', function (event) {
-            var $this = jQuery(this); //the link clicked to make default image
+      //reset selections
 
-            event.preventDefault();
+      //remove hidden from the links
+      jQuery('[data-role="set_as_default_thumbnail"]').removeClass('hidden');
+      //add hidden to the default thumbnail text
+      jQuery('.is_default_thumbnail').addClass('hidden');
+      //reset teh chcked default thumbnail
+      jQuery('input[type="radio"][name="observation[thumb_image_id]"]').prop('checked', false);
 
-            //reset selections
 
-            //remove hidden from the links
-            jQuery('[data-role="set_as_default_thumbnail"]').removeClass('hidden');
-            //add hidden to the default thumbnail text
-            jQuery('.is_default_thumbnail').addClass('hidden');
-            //reset teh chcked default thumbnail
-            jQuery('input[type="radio"][name="observation[thumb_image_id]"]').prop('checked', false);
+      //set selections
 
+      //add hidden to the link clicked
+      $this.addClass('hidden');
+      //show that the image is default
+      $this.siblings('.is_default_thumbnail').removeClass('hidden');
+      //adjust hidden radio button to select default thumbnail
+      $this.siblings('input[type="radio"][name="observation[thumb_image_id]"]').prop('checked', true);
 
-            //set selections
 
-            //add hidden to the link clicked
-            $this.addClass('hidden');
-            //show that the image is default
-            $this.siblings('.is_default_thumbnail').removeClass('hidden');
-            //adjust hidden radio button to select default thumbnail
-            $this.siblings('input[type="radio"][name="observation[thumb_image_id]"]').prop('checked', true);
+    });
 
+    // Detect when a user submits observation; includes upload logic
 
-        });
-
-        // Detect when a user submits observation; includes upload logic
+    $form.submit(function (event) {
+      //  event.preventDefault();
+      if (blockFormSubmission) {
+        fileStore.uploadAll();
+        return false;
+      }
+      return true;
+    });
+  }
+
+  return {
+    init: init
+  }
+}
 
-        $form.submit(function (event) {
-          //  event.preventDefault();
-            if (blockFormSubmission) {
-                fileStore.uploadAll();
-                return false;
-            }
-            return true;
-        });
-    }
 
-    return {
-        init: init
-    }
-}
+jQuery(document).ready(function () {
```

Comment:
Check this SO answer here: https://stackoverflow.com/a/1127864/3357635. I believe using a javascript js.erb file allows you to embed dynamic ruby in there, but of course this needs to be tested. 

> the javascript that is getting returned to the browser can be first rendered by Erb, allowing you to insert custom strings/variables/etc just like you can in .html.erb templates.

Reply:
Oh i get what you're saying. So by making multi_image_upload.js an erb file, i was taking it out of the precompiler, which is not good because look at this thing, it's a huge chunk of js. 

Got it. Will remove the initializer here back to the view.

Reply:
Actually, it was already a `js.erb` file, because it pulls the Ruby config `MO.image_upload_max_size`.
So I guess it wasn't getting precompiled anyway! 

I should maybe fix it so that the page template prints the `MO.image_upload_max_size` as a data-attribute somewhere, and then this JS file can grab it and not have to be parsed by ERB. 

What do you think of that approach?

Reply:
... and that explains why the view has to manually include the upload js in the view — it's not precompiled:
```ruby
<% javascript_include("multi_image_upload") %>
```
I tried deleting that and it didn't load, i was mystified. Now we know.

Reply:
LOL. That's why these PRs take so long... i'm laying those clockwork traps.

Far too many of my cunning plans go undiscovered, though.

Reply:
All `.js.erb`s have now been converted to`.js` and added to the requires in `application.js` 

Most of the time the only Ruby was the asset url for the spinning "indicator.gif". I replaced it with a new CSS spinner in almost all cases.